docs/assets/js/search.js

```diff
@@ -9,43 +9,56 @@ const searchResultsElm = document.getElementById( 'search-results' );
  * Update page markup with search results.
  * @param {Array} results - A list of search result hits as objects.
  * @param {Object} store - search index/meta data store in the window object.
+ * @returns {boolean} True if there are search results, false otherwise.
  */
 function displaySearchResults( results, store ) {
   // Are there any results?
-  if ( results.length ) {
-    let appendString = '';
-
-    // Iterate over the results.
-    for ( let i = 0; i < results.length; i++ ) {
-      const item = store[results[i].ref];
-      appendString += '<li><a href="../' + item.url + '"><h3>' + item.title + '</h3></a>';
-
-      // Show some preview text under each search results item.
-      let previewText = '';
-      const searchMatchWordFragment = Object.keys( results[i].matchData.metadata )[0];
-      const searchMatchFields = results[i].matchData.metadata[searchMatchWordFragment];
-
-      // Remove fields that should never appear as the preview.
-      delete searchMatchFields.id;
-      delete searchMatchFields.title;
+  if ( results.length === 0 ) {
+    displayNoSearchResults( searchResultsElm, results.searchTerm );
+    return false;
+  }
 
-      previewText = item[Object.keys( searchMatchFields )[0]];
+  // Search results content will be placed in here.
+  let resultsString = `
+    <p>
+      ${ results.length } result${ results.length > 1 ? 's' : '' }
+      for search of '${ results.searchTerm }'.
+    </p>
+  `;
+
+  // Iterate over the results.
+  for ( let i = 0; i < results.length; i++ ) {
+    const item = store[results[i].ref];
+    resultsString += '<li><a href="../' + item.url + '"><h3>' + item.title + '</h3></a>';
+
+    // Show some preview text under each search results item.
+    let previewText = '';
+    const searchMatchWordFragment = Object.keys( results[i].matchData.metadata )[0];
+    const searchMatchFields = results[i].matchData.metadata[searchMatchWordFragment];
+
+    // Remove fields that should never appear as the preview.
+    delete searchMatchFields.id;
+    delete searchMatchFields.title;
+
+    const matchFieldKeys = Object.keys( searchMatchFields );
+    if ( matchFieldKeys.length > 0 ) {
+      previewText = item[matchFieldKeys[0]];
+    }
 
-      const regex = new RegExp( results.searchTerm, 'gi' );
-      previewText = previewText.replace( regex, function replace( match ) {
-        return '<mark>' + match + '</mark>';
-      } );
+    const regex = new RegExp( results.searchTerm, 'gi' );
+    previewText = previewText.replace( regex, function replace( match ) {
+      return '<mark>' + match + '</mark>';
+    } );
 
-      // Add the preview text.
-      if ( previewText !== '' ) {
-        appendString += '<p>' + previewText.substring( 0, 150 ) + '…</p></li>';
-      }
+    // Add the preview text.
+    if ( previewText !== '' ) {
+      resultsString += '<p>' + previewText.substring( 0, 150 ) + '…</p></li>';
```

Comment:
Is this a use case for `text-overflow: ellipsis`?

Reply:
Maaaaybe. So here's the trade-off. With `text-overflow` all the preview text would be the same width, which might look nice. It will be only one line though, because to get `text-overflow: ellipsis` to work I have to set the text to not wrap. Also, with `text-overflow`, all the text gets written to the DOM (unless we substring it at some intermediate amount) and then it gets hidden by the CSS, so it's probably more intensive in re-draw. Which do you prefer?

(A)

<img width="1031" alt="Screen Shot 2020-05-08 at 10 16 07 AM" src="https://user-images.githubusercontent.com/704760/81414630-453a2280-9115-11ea-9d97-cb338b15fddd.png">

(B)

<img width="1035" alt="Screen Shot 2020-05-08 at 10 16 57 AM" src="https://user-images.githubusercontent.com/704760/81414641-4b300380-9115-11ea-9258-c54ade8013cd.png">

Reply:
Chat in MM is to keep is (B). An advantage with that is we need no CSS.